imap_processing/cdf/utils.py

```
@@ -102,15 +97,18 @@ def write_cdf(
     dt64 = TTJ2000_EPOCH + dataset["epoch"].values[0].astype("timedelta64[ns]")
     start_time = np.datetime_as_string(dt64, unit="D").replace("-", "")
 
-    # Will now accept vXXX or XXX formats, as batch starter sends versions as vXXX.
-    r = re.compile(r"v\d{3}")
-    if (
-        not isinstance(dataset.attrs["Data_version"], str)
-        or r.match(dataset.attrs["Data_version"]) is None
-    ):
-        version = f"v{int(dataset.attrs['Data_version']):03d}"  # vXXX
-    else:
-        version = dataset.attrs["Data_version"]
+    version = dataset.attrs.get("Data_version", None)
+    if version is None:
+        warnings.warn(
+            "No Data_version attribute found in dataset. Using default v001.",
+            stacklevel=2,
+        )
+        version = "v001"
```

I don't know if we need to but should we raise value missing error instead of default to something?

---

I would suggest defaulting to something that's clearly not a standard version (maybe v999 or v000)

---

Great suggestions, thank you!

I just pushed a new commit with v999. That seemed more anomalous to me than 000, although it might also mean that if we do push up files with that default then we will always see that as the latest. I can change to 000 if we'd prefer that as well.

---

We do need this as a warning because our tests use `write_cdf()` pretty frequently without having a version within the dataset.

I was actually thinking that might have been a better way to go initially would be to create an "empty" dataset in preprocessing for every instrument that has everything they will need for attributes already on it. But, I think that would be yet another major refactor/lift and probably much harder than this one.

---

Ah - good point about that showing up as the latest... I think that's still fine, but I wonder if `v-999` or similar is an allowable value?

If not, 999 works!

---

unfortunately not :( Our version checking is too strict! Would have been nice to have some kind of preliminary flag or something for exactly this situation though I agree.